recursion/lesson_plan.md

```diff
@@ -38,16 +52,48 @@ About ___ hours
 
 ### Things to Remember
 - You can solve all recursion problems with a while loop and vice versa 
-        - While loop solutions usually run faster and take up less computer memory
```

Comment:
This isn't true for programming languages with `Tail Call Optimization`.  Since I don't think this should get into tail call optimization in these lessons, I don't think we should include this in the lesson outline.

Reply:
I think this is probably worth leaving in -- even in languages with TCO, this isn't always the case. In any case, since they're only learning non-TCO languages it seems worth keeping. When later in their careers, the apprentices learn about languages with TCO they can learn about exceptions to this rule.